### PR TITLE
#161966617 favorite articles 

### PR DESCRIPTION
**Pull request by salma-nyagaka**
### What does this pull request do?
 This PR implements the functionality of favoriting articles.
### Description of the tasks to be completed?
- favorite article
  ` PUT /api/slug/favorite/`
- unfavorite article
    ` PUT /api/slug/favorite/`

### How should this be manually tested?
       $ git clone https://github.com/andela/ah-technocrats.git
       $ cd ah-technocrats
       $ git checkout ft-favorite-articles-161966617
       $ python3 -m venv venv
       $ source venv/bin/activate
       $ pip install -r requirements.txt
       $ python manage.py makemigrations
       $ python manage.py migrate
       $ python manage.py runserver

testing: 

       $ python manage.py test

#### Favorite an article

screen shot 2018-12-13 at 16 54 37

#### Unfavorite an article
screen shot 2018-12-13 at 16 54 36

### What are the relevant Pivotal Tracker Stories
[#161966617](https://www.pivotaltracker.com/story/show/161966617)
# Checklist:
- [x] My code follows the style guidelines of this project
- [ ] At least 2 people have reviewed my PR
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] My PR has one commit.

### CODE REVIEW

**Comment by SilasKenneth** [comment]
Good work there 👏

File: authors/apps/articles/views.py [hunk 1]
```diff
@@ -423,3 +426,30 @@ def post(self, request, **kwargs):
...
+           }, status.HTTP_404_NOT_FOUND)
+
+        # gets the user of that specific session
+       user = request.user
+
+        # checks for the boolean value of making an article a favorite
+       confirm = bool(user in article.favorite.all())
+       if confirm is True:
+           article.favorite.remove(user.id)
+           message = {"article": "You have removed this article from your favorites"}
```
Comment (Mnickii) [inline authors/apps/articles/views.py]: Better would be message : Success. This article is a favourite

**Review by Mnickii — DISMISSED** [review]
LGTM

File: authors/apps/articles/tests/base_test.py [hunk 2]
```diff
@@ -13,10 +13,16 @@ def setUp(self):
         # urls
         self.register_url = reverse("authentication:user-signup")
         self.login_url = reverse("authentication:user-login")
+        self.register_url = reverse("authentication:user-signup")
```
Comment (Mnickii) [inline authors/apps/articles/tests/base_test.py:14]: Duplicates

File: authors/apps/articles/tests/base_test.py [hunk 3]
```diff
@@ -13,10 +13,16 @@ def setUp(self):
...
+        self.login_url = reverse("authentication:user-login")
         self.articles_url = reverse("articles:articles")
-        
+        # self.article_url = reverse("articles:get_article", kwargs={slug:'slug'})
```
Comment (Mnickii) [inline authors/apps/articles/tests/base_test.py]: remove this commented code

File: authors/apps/articles/tests/base_test.py (hunk 2 above)
Comment (salma-nyagaka) [reply authors/apps/articles/tests/base_test.py:14]: I have resolved this but this  is not getting outdated

**Review by Mnickii — APPROVED** [review]
LGTM

**Pull request merged**